pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -246,13 +247,26 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []model.WorkloadInfo, svcs
 					TCP:  match.ToChain(tcpClusterName),
 					HTTP: match.ToChain(httpClusterName),
 				}))
+				portProtocols[port.Port] = protocol.Unsupported
 			} else if port.Protocol.IsHTTP() {
 				// Otherwise, just insert HTTP/TCP
 				chains = append(chains, httpChain)
 				portMapper.Map[portString] = match.ToChain(httpChain.Name)
+				// TCP and HTTP on the same port, mark it as requiring sniffing
+				if portProtocols[port.Port] == protocol.TCP {
```

Comment:
So if we have something else that is unsupported, and then we see HTTP/TCP that's fine?

Reply:
good call, I think thats not quite right